### PR TITLE
Enh: add hatch_vcs for versioning

### LINKED ISSUE #82: Migrate to VCS-based versioning using hatchling
Rather than static versioning, we should migrate pyosmeta to hatch_vcs for versioning and setup a GitHub action to trigger a release to pypi upon github release

* this also can be used as a practice run for our python packaging tutorial on this topic.

- [x] First, add vcs versioning to repo (see #85 )
- [x] #100

### PR DESCRIPTION
**Pull request by lwasser**
this is a start at vcs based versioning. i opened a question around how to [setup the correct template here](https://github.com/ofek/hatch-vcs/issues/60) i'm sure that i just don't understand how to help it understand how we are versioning things. right now it's grabbing version. 0.1 even tho we have tag 0.2

addresses in part #82

### CODE REVIEW

**Comment by lwasser** [comment]
something is odd because i see the tags in the history but there are no tags listed here in the repo.

File: src/pyosmeta/_version.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# file generated by setuptools_scm
+# don't change, don't track in version control
```
Comment (ofek) [inline src/pyosmeta/_version.py:2]: This should go in `.gitignore`

Comment (lwasser) [reply src/pyosmeta/_version.py]: Thank you so so much, @ofek !! i just fixed this and pushed it up. i really appreciate your help.

**Comment by lwasser** [comment]
@all-contributors please add @ofek for code, review

**Comment by lwasser** [comment]
This is now ready for review! everything is working as it should locally now 🚀 and i've cleaned up the pyproject.toml file thanks to @ofek suggestion above! ✨

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -188,8 +188,59 @@ actions located here](https://github.com/pyOpenSci/pyopensci.github.io/tree/main
...
+```
+# Install the package and dev dependencies
+pip install -e ".[dev]"
+# Build the package distribution files
+hatch build
+```
+
+When you install dependencies above, `hatch` will be
+installed. To see what version of the package is being
+build run:
```
Comment (ofek) [inline CONTRIBUTING.md:224]:
```suggestion
built run:
```

**Review by ofek — APPROVED** [review]

**Comment by lwasser** [comment]
pre-commit.ci autofix

**Comment by lwasser** [comment]
thank you again @ofek im going to merge this now!!

**Pull request merged**